pkg/component/component.go

```diff
@@ -1560,34 +1562,62 @@ func getRemoteComponentMetadata(client *occlient.Client, componentName string, a
 		}
 	}
 
-	linkedServices := make([]string, 0, 5)
-	linkedComponents := make(map[string][]string)
-	linkedSecretNames := fromCluster.GetLinkedSecretNames()
-	for _, secretName := range linkedSecretNames {
-		secret, err := client.GetKubeClient().GetSecret(secretName, projectName)
+	ok, err := client.GetKubeClient().IsServiceBindingSupported()
+	if err != nil {
+		return Component{}, fmt.Errorf("unable to check if service binding is supported: %w", err)
+	}
+	if ok {
+		linkedSecrets := fromCluster.GetLinkedSecrets()
+		err = setLinksServiceNames(client, linkedSecrets)
 		if err != nil {
-			return Component{}, errors.Wrapf(err, "unable to get info about secret %s", secretName)
-		}
-		componentName, containsComponentLabel := secret.Labels[componentlabels.ComponentLabel]
-		if containsComponentLabel {
-			if port, ok := secret.Annotations[kclient.ComponentPortAnnotationName]; ok {
-				linkedComponents[componentName] = append(linkedComponents[componentName], port)
-			}
-		} else {
-			linkedServices = append(linkedServices, secretName)
+			return Component{}, fmt.Errorf("unable to get name of services: %w", err)
 		}
+		component.Status.LinkedServices = linkedSecrets
 	}
 
 	component.Namespace = client.Namespace
 	component.Spec.App = applicationName
 	component.Spec.Env = filteredEnv
-	component.Status.LinkedComponents = linkedComponents
-	component.Status.LinkedServices = linkedServices
 	component.Status.State = StateTypePushed
 
 	return component, nil
 }
 
+// setLinksServiceNames sets the service name of the links from the info in ServiceBindingRequests present in the cluster
+func setLinksServiceNames(client *occlient.Client, linkedSecrets []SecretMount) error {
+	serviceBindings := map[string]string{}
+	list, err := client.GetKubeClient().ListDynamicResource(kclient.ServiceBindingGroup, kclient.ServiceBindingVersion, kclient.ServiceBindingResource)
+	if err != nil || list == nil {
+		return err
+	}
+	for _, u := range list.Items {
+		var sbr servicebinding.ServiceBinding
+		js, err := u.MarshalJSON()
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(js, &sbr)
+		if err != nil {
+			return err
+		}
+		services := sbr.Spec.Services
+		if len(services) != 1 {
+			return errors.New("ServiceBinding should have only one service")
```

Comment:
I am curious, when is this error raised?

Reply:
In normal operation, it shouldn't. But if someone edits the devfile to add a Service in the Kubernetes inline definition of the Service Binding, it will error.